### ISSUE TITLE
v0.13.5 -no-upgrade build now requires gcc

### ISSUE THREAD

**Issue by siomiz** [syncthing/syncthing#3262]
### Version information

Syncthing Version: v0.13.5
OS Version: Alpine Linux on Docker (`golang:1.6-alpine`)
Browser Version: N/A

### What happened 

Building v0.13.5 by `go run build.go -no-upgrade` returns an error unless `gcc` is installed (but working `bin/syncthing` is produced),
but build.go runs successfully without `gcc` if `-no-upgrade` is removed.

### What I expected

The same result as v0.13.4, i.e. no error from `build.go`
or at least the change in prerequisites mentioned in the changelog.

### Steps to reproduce

```
$ docker run -it --rm golang:1.6-alpine sh
/go # apk add -U git
/go # mkdir -p src/github.com/syncthing/syncthing
/go # cd src/github.com/syncthing/syncthing
/go/src/github.com/syncthing # git clone https://github.com/syncthing/syncthing .
/go/src/github.com/syncthing # cd syncthing
/go/src/github.com/syncthing/syncthing # git checkout v0.13.5
/go/src/github.com/syncthing/syncthing # go run build.go -no-upgrade
```
_Edit: Added missing `cd`_

**Comment by calmh** [comment]
Yeah it's a tag problem somewhere. Will fix once I'm back on a computer on Monday.

**Comment by siomiz** [comment]
@calmh Thank you for looking into this!
It seems `cznic/zappy` wants to go `cgo` for a `-no-upgrade` `discosrv` build.

**Comment by calmh** [comment]
Yes. We disable that with the `purego` build tag but apparently that gets overridden in the noupgrade case. I'll fix it for 0.13.6 anyway.
